src/Microsoft.Azure.EventHubs/EventHubClient.cs

```diff
@@ -599,10 +580,9 @@ internal EventDataSender CreateEventSender(string partitionId = null)
         /// </summary>
         protected override void OnRetryPolicyUpdate()
         {
-            // Propagate retry policy updates to inner sender if there is any.
-            if (this.innerSender != null)
+            if (this.innerSender.IsValueCreated)
```

Comment:
>this.innerSender.IsValueCreated [](start = 16, length = 31)

This defeats the lazy purpose and creates innerSender. innserSender should only be created for 'Send' purpose only.

Reply:
You are so right. I seemed to have missed lazy initialization note.

---
In reply to: [232746929](https://github.com/Azure/azure-event-hubs-dotnet/pull/345#discussion_r232746929) [](ancestors = 232746929)